src/Sulu/Bundle/WebsiteBundle/Controller/RedirectController.php

```diff
@@ -0,0 +1,143 @@
+<?php
+/*
+ * This file is part of Sulu.
+ *
+ * (c) MASSIVE ART WebServices GmbH
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace Sulu\Bundle\WebsiteBundle\Controller;
+
+use Symfony\Bundle\FrameworkBundle\Controller\Controller;
+use Symfony\Component\HttpFoundation\RedirectResponse;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpKernel\Exception\HttpException;
+use Symfony\Component\Routing\Generator\UrlGeneratorInterface;
+
+/**
+ * Contains redirect actions.
+ */
+class RedirectController extends Controller
+{
+    /**
+     * Creates a redirect for configured webspaces.
+     *
+     * @param Request $request
+     *
+     * @return RedirectResponse
+     */
+    public function redirectWebspaceAction(Request $request)
+    {
+        $url = $this->resolveRedirectUrl(
+            $request->get('redirect'),
+            $request->getUri()
+        );
+
+        return new RedirectResponse($url, 301);
+    }
+
+    /**
+     * Creates a redirect for *.html to * (without html).
+     *
+     * @param Request $request
+     *
+     * @return RedirectResponse
+     */
+    public function redirectAction(Request $request)
+    {
+        return new RedirectResponse($request->get('url'), 301);
+    }
+
+    /**
+     * Create a redirect response which uses a route to generate redirect.
+     *
+     * @param Request $request
+     * @param string $route
+     * @param bool $permanent
+     *
+     * @return RedirectResponse
+     */
+    public function redirectToRouteAction(Request $request, $route, $permanent = false)
+    {
+        if ('' === $route) {
+            throw new HttpException($permanent ? 410 : 404);
```

Comment:
I think 410 is not a good idea here. Indicates that there was a resource, but is not anymore, which is not the case here. Why do you want to differentiate at all?